stackrox/stackrox
# ROX-11018: Add option to make auth providers unmodifiable

## Description

This PR adds the option to create unmodifiable auth providers in Central. This is done to prevent ACSMS users from accidentally logging out of their ACS instance by modifying or deleting the `sso.redhat.com` auth provider.
Within this PR `unmodifiable` field is added to auth provider proto. 
When `unmodifiable` set to `true`, it's impossible to:
* modify auth provider
* delete auth provider via UI
* delete auth provider via API until `force` query parameter is set to `true`. 

`force` is a temporary parameter which is introduced so in worst case scenario it's possible to fix authentication within Managed Central.  It will be removed after we switch to declarative configuration for `sso.redhat.com` auth provider

## Checklist

If any of these don't apply, please comment below.

## Testing Performed

#### Manual tests
1. Create unmodifiable auth provider -- verify the configuration is non-editable in the UI, but groups can be changed
2. Create unmodifiable auth provider -- verify it's impossible to delete from UI
3.  Create unmodifiable auth provider -- verify it's impossible to delete from API until `force` is set to `true`
4. Create unmodifiable auth provider -- verify it's impossible to `PUT`/`PATCH` auth provider
